src/snmpd/patch-5.7.3+dfsg/0006-Release-all-requests-that-use-this-session.patch

```diff
@@ -0,0 +1,26 @@
+From 84846206c7ee230bd7b6274af98513952c4a7a7f Mon Sep 17 00:00:00 2001
+From: Renuka Manavalan <remanava@microsoft.com>
+Date: Wed, 7 Aug 2019 21:48:33 +0000
+Subject: [PATCH] Release all requests that use this session.
+
+---
+ agent/snmp_agent.c | 3 ++-
+ 1 file changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/agent/snmp_agent.c b/agent/snmp_agent.c
+index b96d650..ee3b0da 100644
+--- a/agent/snmp_agent.c
++++ b/agent/snmp_agent.c
+@@ -1542,7 +1542,8 @@ netsnmp_remove_delegated_requests_for_session(netsnmp_session *sess)
+          * check each request
+          */
+         netsnmp_request_info *request;
+-        for(request = asp->requests; request; request = request->next) {
++        int i;
++        for(i = 0, request = asp->requests; i < asp->vbcount; ++i, ++request) {
```

Comment:
Do we need to go to the next request here by 
`request = request->next`
Otherwise I don't know what `++request` is doing

Reply:
I asked Renuka about this offline. Apparently this is not a true linked list. @renukamanavalan can provide more detail.

Reply:
The requests are an array of netsnmp_request_info structs allocated.
Each is initialized for a variable in pdu->variables
Each request is assigned a subtree based on variable->name
Each netsnmp_subtree has its cache_id
The requests that carry same cache-id through their netsnmp_subtree are linked while assigning to tree cache.

In short, the requests are not maintained as link list, but linked while grouping per cache-id.

So when you are scanning across all, walk from 0 to vbcount.